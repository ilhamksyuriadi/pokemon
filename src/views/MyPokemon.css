.list-layout{
    width: 100vw;
    max-width: 100%;
    height: 400px;
    display: flex;
}

@media (max-width: 500px) {
    .list-layout{
        flex-direction: column;
        align-items: center;
        height: 400px;
    }
}

@media (min-width: 501px) {
    .list-layout{
        justify-content: space-around;
        align-items: center;
        flex-wrap: wrap;
    }
}