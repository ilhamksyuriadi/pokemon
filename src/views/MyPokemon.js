import React, { useState } from 'react';
import getMyPokemon from '../operations/queries/getMyPokemon';
import CardList from '../components/CardList';
import releasePokemon from '../operations/mutations/deletePokemon';

const MyPokemon = () => {
    const [pokemonsState, setPokemonsState] = useState(getMyPokemon())

    const handleRelease = (e, nickname) => {
        e.stopPropagation()
        releasePokemon(nickname)
        setPokemonsState(getMyPokemon())
    }

    const pokemons = pokemonsState.map(({id, name, nickname}) => {
        const imageUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`
        return (
            <CardList 
                key={nickname}
                name={name}
                image={imageUrl}
                nickname={nickname}
                onRelease={handleRelease}
            />
        )
    })
    return (
        <div className="list-layout">
            {pokemons}
        </div>
    )
}

export default MyPokemon;