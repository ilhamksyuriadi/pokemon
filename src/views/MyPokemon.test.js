import MyPokemon from './MyPokemon';
import { unmountComponentAtNode } from 'react-dom';
import renderer from  'react-test-renderer';

let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("integration - my pokemon, snapshot", () => {
    const tree = renderer.create(<MyPokemon />).toJSON();
    expect(tree).toMatchSnapshot();
})