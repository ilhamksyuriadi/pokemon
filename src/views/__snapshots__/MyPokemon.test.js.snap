// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration - my pokemon, snapshot 1`] = `
Array [
  <h2
    className="title"
  >
    Catched Pokemon List
  </h2>,
  <code>
    Total: 
    0
  </code>,
  <div
    className="empty-layout"
  >
    <div>
      <img
        alt="pidgey"
        src="pidgey-back.png"
      />
      <h4>
        you don't have any pokemon yet
      </h4>
    </div>
  </div>,
]
`;
